cmd/deploy/deploy.go

```diff
@@ -611,14 +623,7 @@ func buildImages(ctx context.Context, build func(context.Context, *types.BuildOp
 		// 2. deployOptions.servicesToDeploy
 		// For that we'll need to know which services were defined in the stack, to know which ones were defined in the okteto manifest
 		// The `Manifest` has all the services combined into one, so we need (allServices - stackServices) + deployOptions.servicesToDeploy
-		var stackServices map[string]bool
-
-		if stack := deployOptions.Manifest.GetStack(); stack != nil {
-			stackServices = stack.GetServicesWithBuildSection()
-		}
 
-		allServicesWithBuildSection := deployOptions.Manifest.GetBuildServices()
-		oktetoManifestBuildServices := setDifference(allServicesWithBuildSection, stackServices)
 		// servicesToBuildSet = (allServicesWithBuildSection - stackServices) + deployOptions.servicesToDeploy
 		servicesToBuildSet := setUnion(oktetoManifestBuildServices, sliceToSet(deployOptions.servicesToDeploy))
```

Comment:
Based on the new logic that I'm seeing, I think that we should apply the same here: only account for services which have a build section

Reply:
I see the only thing change here is common logic for both build and not build flag has been taken out the if/else. What do you mean with `apply the same here?`

Reply:
I think that in both cases (build forced and unforced) we should be collecting the same images. The only difference would be in the unforced build that we then filter that slice of images if they have been already built

So I think we should do here the same we are doing in the other block 

```
finalStackIntersection := setIntersection(stackServices, sliceToSet(deployOptions.servicesToDeploy))
servicesToBuildSet := setUnion(oktetoManifestBuildServices, finalStackIntersection)
```